From 1a9ee9df4ff27b6c96b2a9b76519ae0b2d936ab7 Mon Sep 17 00:00:00 2001
From: Suneel Garapati <sgarapati@caviumnetworks.com>
Date: Tue, 25 Jul 2017 18:41:17 -0700
Subject: [PATCH 0659/1239] config: kconfig changes for t81/83/93

u-boot has moved to kconfig(not completely), this is first
commit to cleanup kconfig related to board specific code and
legacy include files.
More kconfig cleanup is required, also v2017.09 should have
complete kconfig menu entries.

Signed-off-by: Suneel Garapati <sgarapati@caviumnetworks.com>
---
 arch/arm/Kconfig                |  12 ++++
 arch/arm/mach-thunderx/Kconfig  |  88 ------------------------
 board/cavium/thunderx/Kconfig   |  16 -----
 configs/t83_defconfig           |  48 +++++++------
 configs/t93_defconfig           |  46 ++++++++-----
 include/configs/thunderx_81xx.h |  93 +-------------------------
 include/configs/thunderx_83xx.h | 108 +-----------------------------
 include/configs/thunderx_93xx.h | 115 +-------------------------------
 8 files changed, 75 insertions(+), 451 deletions(-)

diff --git a/arch/arm/Kconfig b/arch/arm/Kconfig
index c1f57207f3..49d6b46783 100644
--- a/arch/arm/Kconfig
+++ b/arch/arm/Kconfig
@@ -1598,6 +1598,18 @@ config ARCH_ROCKCHIP
 	imply TPL_SYSRESET
 	imply USB_FUNCTION_FASTBOOT
 
+config ARCH_THUNDERX
+	bool "Support Cavium ThunderX SoCs"
+	select ARM64
+	select SYS_CACHE_SHIFT_7
+	select OF_CONTROL
+	select OF_SEPARATE
+	select OF_LIBFDT
+	select DM
+	select DM_SERIAL
+	select DM_PCI
+	select DM_PCI_COMPAT
+
 config TARGET_THUNDERX_88XX
 	bool "Support ThunderX 88xx"
 	select ARM64
diff --git a/arch/arm/mach-thunderx/Kconfig b/arch/arm/mach-thunderx/Kconfig
index ddaf5ee083..3f685b09bf 100644
--- a/arch/arm/mach-thunderx/Kconfig
+++ b/arch/arm/mach-thunderx/Kconfig
@@ -1,45 +1,5 @@
 if ARCH_THUNDERX
 
-config	THUNDERX_COMMON
-	bool
-	default y
-	select DM
-	select DM_I2C
-	select DM_I2C_COMPAT
-	select DM_GPIO
-	select DM_PCI
-	select DM_PCI_COMPAT
-	select DM_SERIAL
-	select DM_USB
-	select DM_MMC
-	select OF_CONTROL
-	select OF_EMBED
-	select OF_LIBFDT
-	select ARM64
-	select PCI_THUNDERX_ECAM
-	select PL01X_SERIAL
-	select SYS_64BIT
-	select NO_RELOCATION
-	select PCI
-	select SYS_PCI_64BIT
-	select PCI_PNP
-	select PCI_AUTO
-	select CAVIUM_MMC
-	select MMC
-
-choice
-	prompt "Choose target processor"
-
-	config TARGET_THUNDERX_CN81XX
-		bool "ThunderX CN81XX family"
-		select SYS_SOC_THUNDERX_81XX
-
-	config TARGET_THUNDERX_CN88XX
-		bool "ThunderX CN88XX family"
-		select SYS_SOC_THUNDERX_88XX
-
-endchoice
-
 config PCI_AUTO
 	bool
 	default y
@@ -48,10 +8,6 @@ config OF_LIBFDT
 	bool
 	default y
 
-config SYS_ARCH
-	string
-	default "thunderx"
-
 config SYS_VENDOR
 	string
 	default	"cavium"
@@ -98,26 +54,6 @@ config SYS_64BIT
 	bool
 	default y
 
-config SYS_VA_BITS
-	int
-	default 48
-
-config SYS_PTL2_BITS
-	int
-	default 42
-
-config SYS_BLOCK_SHIFT
-	int
-	default 29
-
-config SYS_PTL1_ENTRIES
-	int
-	default 64
-
-config SYS_PTL2_ENTRIES
-	int
-	default 8192
-
 config NO_RELOCATION
 	bool
 	default y
@@ -134,10 +70,6 @@ config PCI_PNP
 	bool
 	default y
 
-config SYS_PCI_ARI
-	bool
-	default y
-
 config DM_I2C
 	bool
 	default y
@@ -158,34 +90,16 @@ config DM
 	bool
 	default y
 
-
-config MMC
-	bool
-	default y
-
 config CAVIUM_MMC
 	bool
 	default y
 	select DM_MMC
-	select MMC
 	select FS_EXT4
 
 config OF_CONTROL
 	bool
 	default y
 
-config CMD_ATF
-	bool "Enable ATF query commands"
-	default y
-	help
-	 Enable vendor specific ATF query commands such as SPI and SD/MMC
-	 devices access, low level environment query, boot device layout
-	 and node count.
-
-config SYS_CACHELINE_SIZE
-	int
-	default 128
-
 config PCI_THUNDERX_ECAM
 	bool
 	default y
@@ -195,6 +109,4 @@ config PCI_THUNDERX_ECAM
 	 Enhanced Configuration Access Mechanism (ECAM), and PCI configuration
 	 blocks for ThunderX on-chip functions.
 
-#source "arch/arm/mach-octeontx/cn81xx/Kconfig"
-
 endif
diff --git a/board/cavium/thunderx/Kconfig b/board/cavium/thunderx/Kconfig
index d87bf580f1..8e635ad63a 100644
--- a/board/cavium/thunderx/Kconfig
+++ b/board/cavium/thunderx/Kconfig
@@ -1,4 +1,3 @@
-
 choice
 	prompt "Target board"
 
@@ -8,9 +7,6 @@ config	TARGET_THUNDERX_83XX
 config	TARGET_THUNDERX_81XX
 	bool "ThunderX CN81XX"
 
-config	TARGET_THUNDERX_88XX
-	bool "ThunderX CN88XX"
-
 config	TARGET_THUNDERX_93XX
 	bool "ThunderX CN93XX"
 
@@ -20,18 +16,6 @@ config SYS_BOARD
 	string
 	default "thunderx"
 
-config DM_I2C
-	bool
-	default y
-
-config DM_I2C_COMPAT
-	bool
-	default y
-
-config DM_GPIO
-	bool
-	default y
-
 if TARGET_THUNDERX_88XX || TARGET_THUNDERX_81XX || TARGET_THUNDERX_83XX || TARGET_THUNDERX_93XX
 
 config CMD_ATF
diff --git a/configs/t83_defconfig b/configs/t83_defconfig
index c41668cf72..09f759572d 100644
--- a/configs/t83_defconfig
+++ b/configs/t83_defconfig
@@ -1,17 +1,20 @@
 CONFIG_ARM=y
-CONFIG_THUNDERX_COMMON=y
-#CONFIG_SYS_SOC="thunderx"
+# CONFIG_ARM64_SUPPORT_AARCH32 is not set
 CONFIG_ARCH_THUNDERX=y
-CONFIG_TARGET_THUNDERX_83XX=y
-CONFIG_DM_I2C=y
-CONFIG_DM_GPIO=y
 CONFIG_DEFAULT_DEVICE_TREE="thunder-83xx"
+CONFIG_DEBUG_UART=y
+CONFIG_FIT=y
+CONFIG_FIT_SIGNATURE=y
 CONFIG_SYS_EXTRA_OPTIONS="ARM64"
+CONFIG_BOOTDELAY=5
+# CONFIG_DISPLAY_CPUINFO is not set
 CONFIG_HUSH_PARSER=y
-CONFIG_SYS_PROMPT="ThunderX_83XX> "
 CONFIG_CMD_MEMTEST=y
 CONFIG_CMD_MX_CYCLIC=y
 # CONFIG_CMD_FLASH is not set
+CONFIG_CMD_MMC=y
+CONFIG_CMD_PART=y
+CONFIG_CMD_SF=y
 CONFIG_CMD_I2C=y
 CONFIG_CMD_USB=y
 # CONFIG_CMD_FPGA is not set
@@ -25,29 +28,36 @@ CONFIG_CMD_CDP=y
 CONFIG_CMD_SNTP=y
 CONFIG_CMD_DNS=y
 CONFIG_CMD_LINK_LOCAL=y
+CONFIG_CMD_EXT4=y
+CONFIG_CMD_EXT4_WRITE=y
+CONFIG_CMD_FAT=y
+CONFIG_ISO_PARTITION=y
+CONFIG_EFI_PARTITION=y
+CONFIG_PARTITION_TYPE_GUID=y
+CONFIG_OF_BOARD=y
 CONFIG_NET_RANDOM_ETHADDR=y
-CONFIG_DM_I2C_COMPAT=y
 CONFIG_MISC=y
 CONFIG_PHYLIB=y
+CONFIG_PHYLIB_10G=y
+CONFIG_PHY_AQUANTIA=y
+CONFIG_PHY_BROADCOM=y
+CONFIG_PHY_MARVELL=y
+CONFIG_PHY_MICREL=y
+CONFIG_PHY_MICREL_KSZ9021=y
+CONFIG_PHY_MICREL_KSZ9031=y
+CONFIG_PHY_MSCC=y
+CONFIG_PHY_REALTEK=y
+CONFIG_PHY_VITESSE=y
 CONFIG_NETDEVICES=y
-CONFIG_PCI=y
-CONFIG_PCI_PNP=y
-CONFIG_PCI_THUNDERX_ECAM=y
-CONFIG_DEBUG_UART=y
+CONFIG_DM_RTC=y
 CONFIG_DEBUG_UART_PL011=y
 CONFIG_DEBUG_UART_BASE=0x87e028000000
 CONFIG_DEBUG_UART_CLOCK=24000000
 CONFIG_DEBUG_UART_SKIP_INIT=y
+CONFIG_DM_SPI=y
+CONFIG_CAVIUM_SPI=y
 CONFIG_USB=y
 CONFIG_DM_USB=y
-CONFIG_DM_MMC=y
-CONFIG_DM=y
 CONFIG_USB_XHCI_HCD=y
 CONFIG_USB_STORAGE=y
-CONFIG_SYS_CONFIG_NAME="thunder_83xx"
-CONFIG_DM_RTC=y
 CONFIG_ERRNO_STR=y
-CONFIG_SYS_I2C_THUNDERX_SLAVE_ADDR=0x77
-CONFIG_ENV_VARS_UBOOT_CONFIG=y
-CONFIG_OF_LIBFDT=y
-CONFIG_OF_BOARD=y
diff --git a/configs/t93_defconfig b/configs/t93_defconfig
index 18696006e6..fabba817fe 100644
--- a/configs/t93_defconfig
+++ b/configs/t93_defconfig
@@ -1,17 +1,21 @@
 CONFIG_ARM=y
-CONFIG_THUNDERX_COMMON=y
-#CONFIG_SYS_SOC="thunderx"
+# CONFIG_ARM64_SUPPORT_AARCH32 is not set
 CONFIG_ARCH_THUNDERX=y
 CONFIG_TARGET_THUNDERX_93XX=y
-CONFIG_DM_I2C=y
-CONFIG_DM_GPIO=y
 CONFIG_DEFAULT_DEVICE_TREE="thunder-93xx"
+CONFIG_DEBUG_UART=y
+CONFIG_FIT=y
+CONFIG_FIT_SIGNATURE=y
 CONFIG_SYS_EXTRA_OPTIONS="ARM64"
+CONFIG_BOOTDELAY=5
+# CONFIG_DISPLAY_CPUINFO is not set
 CONFIG_HUSH_PARSER=y
-CONFIG_SYS_PROMPT="ThunderX_93XX> "
 CONFIG_CMD_MEMTEST=y
 CONFIG_CMD_MX_CYCLIC=y
 # CONFIG_CMD_FLASH is not set
+CONFIG_CMD_MMC=y
+CONFIG_CMD_PART=y
+CONFIG_CMD_SF=y
 CONFIG_CMD_I2C=y
 CONFIG_CMD_USB=y
 # CONFIG_CMD_FPGA is not set
@@ -25,26 +29,38 @@ CONFIG_CMD_CDP=y
 CONFIG_CMD_SNTP=y
 CONFIG_CMD_DNS=y
 CONFIG_CMD_LINK_LOCAL=y
+CONFIG_EFI_PARTITION=y
+CONFIG_PARTITION_TYPE_GUID=y
 CONFIG_NET_RANDOM_ETHADDR=y
-CONFIG_DM_I2C_COMPAT=y
 CONFIG_MISC=y
+CONFIG_SPI_FLASH=y
+CONFIG_SPI_FLASH_MACRONIX=y
+CONFIG_SPI_FLASH_SPANSION=y
+CONFIG_SPI_FLASH_STMICRO=y
+CONFIG_SPI_FLASH_MTD=y
 CONFIG_PHYLIB=y
+CONFIG_PHYLIB_10G=y
+CONFIG_PHY_AQUANTIA=y
+CONFIG_PHY_BROADCOM=y
+CONFIG_PHY_MARVELL=y
+CONFIG_PHY_MICREL=y
+CONFIG_PHY_MICREL_KSZ9021=y
+CONFIG_PHY_MICREL_KSZ9031=y
+CONFIG_PHY_MSCC=y
+CONFIG_PHY_REALTEK=y
+CONFIG_PHY_VITESSE=y
 CONFIG_NETDEVICES=y
-CONFIG_PCI=y
-CONFIG_PCI_PNP=y
-CONFIG_PCI_THUNDERX_ECAM=y
-CONFIG_DEBUG_UART=y
+CONFIG_DM_RTC=y
 CONFIG_DEBUG_UART_PL011=y
 CONFIG_DEBUG_UART_BASE=0x87e028000000
 CONFIG_DEBUG_UART_CLOCK=24000000
 CONFIG_DEBUG_UART_SKIP_INIT=y
+CONFIG_DM_SPI=y
+CONFIG_CAVIUM_SPI=y
 CONFIG_USB=y
 CONFIG_DM_USB=y
 CONFIG_USB_XHCI_HCD=y
 CONFIG_USB_STORAGE=y
-CONFIG_SYS_CONFIG_NAME="thunder_93xx"
-CONFIG_DM_RTC=y
+CONFIG_FS_FAT=y
+CONFIG_FAT_WRITE=y
 CONFIG_ERRNO_STR=y
-CONFIG_SYS_I2C_THUNDERX_SLAVE_ADDR=0x77
-CONFIG_ENV_VARS_UBOOT_CONFIG=y
-CONFIG_OF_LIBFDT=y
diff --git a/include/configs/thunderx_81xx.h b/include/configs/thunderx_81xx.h
index 2968b49335..2724cf5411 100644
--- a/include/configs/thunderx_81xx.h
+++ b/include/configs/thunderx_81xx.h
@@ -21,8 +21,6 @@
 /** Enable vsnprintf support */
 #define CONFIG_SYS_VSNPRINTF
 
-#define CONFIG_BOOTP_VCI_STRING		"Diagnostics"
-
 #define MEM_BASE			0x00500000
 
 #define CONFIG_COREID_MASK             0xffffff
@@ -33,52 +31,12 @@
 /** Low memory base address */
 #define CONFIG_SYS_LOWMEM_BASE		MEM_BASE
 
-/**
- * Memory map for virtual memory.  See arch/arm/include/asm/armv8/mmu.h
- * base:64, size:64, attributes:64
- */
-#define CONFIG_SYS_MEM_MAP		{{0x000000000000UL, 0x40000000000UL, \
-					  PTL2_MEMTYPE(MT_NORMAL) |	     \
-					  PTL2_BLOCK_NON_SHARE},	     \
-					 {0x800000000000UL, 0x40000000000UL, \
-					  PTL2_MEMTYPE(MT_DEVICE_NGNRNE) |   \
-					  PTL2_BLOCK_NON_SHARE},	     \
-					 {0x840000000000UL, 0x40000000000UL, \
-					  PTL2_MEMTYPE(MT_DEVICE_NGNRNE) |   \
-					  PTL2_BLOCK_NON_SHARE},	     \
-					 {0x880000000000UL, 0x40000000000UL, \
-					  PTL2_MEMTYPE(MT_DEVICE_NGNRNE) |   \
-					  PTL2_BLOCK_NON_SHARE},	     \
-					 {0xffffffffffffUL, 0xfffffffffffUL, \
-					  PTL2_MEMTYPE(MT_DEVICE_NGNRNE) |   \
-					  PTL2_BLOCK_NON_SHARE},	     \
-					}
-
-/** Number of entries in CONFIG_SYS_MEM_MAP */
-#define CONFIG_SYS_MEM_MAP_SIZE		4
-
-#define CONFIG_COREID_MASK		0xffffff
-
 /** Number of virtual address bits */
 #define CONFIG_SYS_VA_BITS		48
 
-/** Number of bits addressed by second level */
-#define CONFIG_SYS_PTL2_BITS		42
-
 /** Number of bits addressed by a single block entry from L2 page tables */
 #define CONFIG_SYS_BLOCK_SHIFT		29
 
-/** Number of 1st level page table entries */
-#define CONFIG_SYS_PTL1_ENTRIES		64
-
-/** Number of 2nd level page table entries */
-#define CONFIG_SYS_PTL2_ENTRIES		8192
-
-/** Total size of page table */
-#define CONFIG_SYS_PGTABLE_SIZE		\
-	((CONFIG_SYS_PTL1_ENTRIES + \
-	  CONFIG_SYS_MEM_MAP_SIZE * CONFIG_SYS_PTL2_ENTRIES) * 8)
-
 /** IPS field of TCR_EL1 */
 #define CONFIG_SYS_TCR_EL1_IPS_BITS	(5UL << 32)
 
@@ -90,10 +48,6 @@
 
 #define CONFIG_SLT
 
-/* Enable FIT image support */
-#define CONFIG_FIT
-#define CONFIG_FIT_VERBOSE
-
 /* Link Definitions */
 
 /** Code start address */
@@ -183,12 +137,6 @@
 /** Maximum total number of BGX interfaces across all nodes */
 #define CONFIG_MAX_BGX			3
 
-/** Enable 10G PHY support */
-#define CONFIG_PHYLIB_10G
-#define CONFIG_PHY_VITESSE
-#define CONFIG_PHY_MICREL
-#define CONFIG_PHY_MARVELL
-
 /** Command line configuration */
 #define CONFIG_MENU
 
@@ -201,30 +149,13 @@
 /** Enable env command */
 #define CONFIG_CMD_ENV
 #undef  CONFIG_CMD_IMLS
-#define CONFIG_CMD_BOOTI
 
 #define CONFIG_CMD_MII
 
 /** Enable tftp command */
 #define CONFIG_CMD_TFTP
 
-/** Enable fatload, fatls, etc. */
-#define CONFIG_CMD_FAT
-
-/** Enable write support to FAT partitions */
-#define CONFIG_FAT_WRITE
-
-/** Enable ext2ls, ext2load, etc. commands */
-#define CONFIG_CMD_EXT2
-
-/** Enable ext4ls, ext4load, etc. commands */
-#define CONFIG_CMD_EXT4
-
-/** Enable part command */
-#define CONFIG_CMD_PART
-
 /** Enable sata command */
-#define CONFIG_CMD_SATA
 
 /** Enable pci command */
 #define CONFIG_CMD_PCI
@@ -239,7 +170,6 @@
 #define CONFIG_CMD_GREPENV
 
 /** Enable bootimgup command */
-#define CONFIG_CMD_BOOTIMGUP
 
 /**
  * Enable callback support when environment variables are set, or
@@ -247,15 +177,11 @@
  */
 #define CONFIG_CMD_ENV_CALLBACK
 
-/** Enable date command */
-#define CONFIG_CMD_DATE
-
 #define CONFIG_ENV_CALLBACK_LIST_STATIC "txsmi\\d?mode:smimode"
 
 /* AHCI support Definitions */
 #ifdef CONFIG_CMD_SATA
 /** Enable AHCI SATA driver */
-# define CONFIG_SATA_AHCI
 /** Maximum number of SATA devices */
 # define CONFIG_SYS_SATA_MAX_DEVICE	2
 /** Enable 48-bit SATA addressing */
@@ -266,14 +192,6 @@
 # define CONFIG_SYS_64BIT_LBA
 #endif
 
-/* Partition systems */
-/** Enable old DOS partitioning for drives under 3TB */
-#define CONFIG_DOS_PARTITION
-/** Enable EFI partition support */
-#define CONFIG_EFI_PARTITION
-/** Enable UUIDs for drives >= 3TB */
-#define CONFIG_PARTITION_UUIDS
-
 /** Enable EXT4 filesystem support */
 #define CONFIG_FS_EXT4
 
@@ -298,18 +216,14 @@
 #define CONFIG_SYS_SPD_I2C_BUS 1
 
 /***** SPI Defines *********/
-#define CONFIG_DM_SPI
 #define CONFIG_DM_SPI_FLASH
 #define CONFIG_SPI_FLASH
 #define CONFIG_SPI_FLASH_STMICRO
 #define CONFIG_SPI_FLASH_MACRONIX
-#define CONFIG_CAVIUM_SPI
 #define CONFIG_SF_DEFAULT_SPEED 12500000
 #define CONFIG_SF_DEFAULT_MODE	0
 #define CONFIG_SF_DEFAULT_BUS	0
 #define CONFIG_SF_DEFAULT_CS	0
-/** Enable sf command */
-#define CONFIG_CMD_SF
 
 /**************************/
 #define CONFIG_CMD_SAVES
@@ -320,7 +234,6 @@
 #define CONFIG_BOOTP_GATEWAY
 #define CONFIG_BOOTP_HOSTNAME
 #define CONFIG_BOOTP_PXE
-#define CONFIG_BOOTP_PXE_CLIENTARCH	0x100
 
 /* Miscellaneous configurable options */
 #define CONFIG_SYS_LOAD_ADDR		(MEM_BASE)
@@ -372,16 +285,13 @@
 /** Store U-Boot version in "ver" environment variable */
 #define CONFIG_VERSION_VARIABLE
 
-/** How long to delay before booting */
-#define CONFIG_BOOTDELAY		5
-
 /** Where the environment is located */
 
 /* Use ATF based calls for env variables */
 #define CONFIG_ATF
 
 /* Use uboot SPI APIs for env variables */
-//#define CONFIG_SPI_ENV
+/* #define CONFIG_SPI_ENV */
 
 #if defined(CONFIG_ATF)
 #define CONFIG_ENV_IS_IN_ATF
@@ -433,7 +343,6 @@
 #define NS_PER_REF_CLK_TICK		(1000000000/PLL_REF_CLK)
 
 #define CONFIG_OCTEON_MMC
-#define CONFIG_CMD_MMC
 #define CONFIG_SYS_MMC_MAX_BLK_COUNT	8191
 #define CONFIG_SYS_MMC_SET_DEV
 
diff --git a/include/configs/thunderx_83xx.h b/include/configs/thunderx_83xx.h
index b96465f2c1..0e8989524a 100644
--- a/include/configs/thunderx_83xx.h
+++ b/include/configs/thunderx_83xx.h
@@ -7,10 +7,6 @@
 #ifndef __THUNDERX_83XX_H__
 #define __THUNDERX_83XX_H__
 
-#define CONFIG_SYS_VENDOR	"cavium"
-
-#define CONFIG_SPECIAL_SYNC_HANDLER
-
 /** Thunder 83xx does not support NOR flash */
 #define CONFIG_SYS_NO_FLASH
 
@@ -21,7 +17,6 @@
 /** Enable vsnprintf support */
 #define CONFIG_SYS_VSNPRINTF
 
-#define CONFIG_BOOTP_VCI_STRING		"Diagnostics"
 /* starting uboot and linux kernel from offset 40MB
  * below 40MB will be used for mailboxes of pko/pki
  * */
@@ -36,52 +31,6 @@
 /** Low memory base address */
 #define CONFIG_SYS_LOWMEM_BASE		MEM_BASE
 
-/**
- * Memory map for virtual memory.  See arch/arm/include/asm/armv8/mmu.h
- * base:64, size:64, attributes:64
- */
-#define CONFIG_SYS_MEM_MAP		{{0x000000000000UL, 0x40000000000UL, \
-					  PTL2_MEMTYPE(MT_NORMAL) |	     \
-					  PTL2_BLOCK_NON_SHARE},	     \
-					 {0x800000000000UL, 0x40000000000UL, \
-					  PTL2_MEMTYPE(MT_DEVICE_NGNRNE) |   \
-					  PTL2_BLOCK_NON_SHARE},	     \
-					 {0x840000000000UL, 0x40000000000UL, \
-					  PTL2_MEMTYPE(MT_DEVICE_NGNRNE) |   \
-					  PTL2_BLOCK_NON_SHARE},	     \
-					 {0x880000000000UL, 0x40000000000UL, \
-					  PTL2_MEMTYPE(MT_DEVICE_NGNRNE) |   \
-					  PTL2_BLOCK_NON_SHARE},	     \
-					 {0xffffffffffffUL, 0xfffffffffffUL, \
-					  PTL2_MEMTYPE(MT_DEVICE_NGNRNE) |   \
-					  PTL2_BLOCK_NON_SHARE},	     \
-					}
-
-/** Number of entries in CONFIG_SYS_MEM_MAP */
-#define CONFIG_SYS_MEM_MAP_SIZE		4
-
-#define CONFIG_COREID_MASK		0xffffff
-
-/** Number of virtual address bits */
-#define CONFIG_SYS_VA_BITS		48
-
-/** Number of bits addressed by second level */
-#define CONFIG_SYS_PTL2_BITS		42
-
-/** Number of bits addressed by a single block entry from L2 page tables */
-#define CONFIG_SYS_BLOCK_SHIFT		29
-
-/** Number of 1st level page table entries */
-#define CONFIG_SYS_PTL1_ENTRIES		64
-
-/** Number of 2nd level page table entries */
-#define CONFIG_SYS_PTL2_ENTRIES		8192
-
-/** Total size of page table */
-#define CONFIG_SYS_PGTABLE_SIZE		\
-	((CONFIG_SYS_PTL1_ENTRIES + \
-	  CONFIG_SYS_MEM_MAP_SIZE * CONFIG_SYS_PTL2_ENTRIES) * 8)
-
 /** IPS field of TCR_EL1 */
 #define CONFIG_SYS_TCR_EL1_IPS_BITS	(5UL << 32)
 
@@ -138,9 +87,6 @@
  */
 #define CONFIG_SYS_MEM_TOP_HIDE		0x1000000
 
-/** Console serial port number */
-#define CONFIG_CONS_INDEX		1
-
 /* Generic Interrupt Controller Definitions */
 
 /** Base address of Global Interrupt Controller GICD registers */
@@ -179,12 +125,6 @@
 /** Maximum total number of BGX interfaces across all nodes */
 #define CONFIG_MAX_BGX			4
 
-/** Enable 10G PHY support */
-#define CONFIG_PHYLIB_10G
-#define CONFIG_PHY_VITESSE
-#define CONFIG_PHY_MICREL
-#define CONFIG_PHY_MARVELL
-
 /** Command line configuration */
 #define CONFIG_MENU
 
@@ -197,31 +137,12 @@
 /** Enable env command */
 #define CONFIG_CMD_ENV
 #undef  CONFIG_CMD_IMLS
-#define CONFIG_CMD_BOOTI
 
 #define CONFIG_CMD_MII
 
 /** Enable tftp command */
 #define CONFIG_CMD_TFTP
 
-/** Enable fatload, fatls, etc. */
-#define CONFIG_CMD_FAT
-
-/** Enable write support to FAT partitions */
-#define CONFIG_FAT_WRITE
-
-/** Enable ext2ls, ext2load, etc. commands */
-#define CONFIG_CMD_EXT2
-
-/** Enable ext4ls, ext4load, etc. commands */
-#define CONFIG_CMD_EXT4
-
-/** Enable part command */
-#define CONFIG_CMD_PART
-
-/** Enable sata command */
-#define CONFIG_CMD_SATA
-
 /** Enable pci command */
 #define CONFIG_CMD_PCI
 
@@ -234,18 +155,12 @@
 /** Enable grepenv command */
 #define CONFIG_CMD_GREPENV
 
-/** Enable bootimgup command */
-#define CONFIG_CMD_BOOTIMGUP
-
 /**
  * Enable callback support when environment variables are set, or
  * changed
  */
 #define CONFIG_CMD_ENV_CALLBACK
 
-/** Enable date command */
-#define CONFIG_CMD_DATE
-
 #define CONFIG_ENV_CALLBACK_LIST_STATIC "txsmi\\d?mode:smimode"
 
 /* AHCI support Definitions */
@@ -262,17 +177,6 @@
 # define CONFIG_SYS_64BIT_LBA
 #endif
 
-/* Partition systems */
-/** Enable old DOS partitioning for drives under 3TB */
-#define CONFIG_DOS_PARTITION
-/** Enable EFI partition support */
-#define CONFIG_EFI_PARTITION
-/** Enable UUIDs for drives >= 3TB */
-#define CONFIG_PARTITION_UUIDS
-
-/** Enable EXT4 filesystem support */
-#define CONFIG_FS_EXT4
-
 /* PCIe */
 /** Show devices found on PCI bus */
 #define CONFIG_PCI_SCAN_SHOW
@@ -300,18 +204,15 @@
 #define CONFIG_SYS_SPD_I2C_BUS 1
 
 /***** SPI Defines *********/
-#define CONFIG_DM_SPI
 #define CONFIG_DM_SPI_FLASH
 #define CONFIG_SPI_FLASH
 #define CONFIG_SPI_FLASH_STMICRO
 #define CONFIG_SPI_FLASH_MACRONIX
-#define CONFIG_CAVIUM_SPI
 #define CONFIG_SF_DEFAULT_SPEED 12500000
 #define CONFIG_SF_DEFAULT_MODE	0
 #define CONFIG_SF_DEFAULT_BUS	0
 #define CONFIG_SF_DEFAULT_CS	0
 /** Enable sf command */
-#define CONFIG_CMD_SF
 
 /**************************/
 #define CONFIG_CMD_SAVES
@@ -322,7 +223,6 @@
 #define CONFIG_BOOTP_GATEWAY
 #define CONFIG_BOOTP_HOSTNAME
 #define CONFIG_BOOTP_PXE
-#define CONFIG_BOOTP_PXE_CLIENTARCH	0x100
 
 /* Miscellaneous configurable options */
 #define CONFIG_SYS_LOAD_ADDR		(MEM_BASE)
@@ -374,16 +274,12 @@
 /** Store U-Boot version in "ver" environment variable */
 #define CONFIG_VERSION_VARIABLE
 
-/** How long to delay before booting */
-#define CONFIG_BOOTDELAY		5
-
-/** Where the environment is located */
 
 /* Use ATF based calls for env variables */
 #define CONFIG_ATF
 
 /* Use uboot SPI APIs for env variables */
-//#define CONFIG_SPI_ENV
+/*#define CONFIG_SPI_ENV*/
 
 #if defined(CONFIG_ATF)
 #define CONFIG_ENV_IS_IN_ATF
@@ -427,7 +323,6 @@
 #define CONFIG_AUTO_COMPLETE
 #define CONFIG_FEATURE_COMMAND_EDITING
 #define CONFIG_SYS_MAXARGS		64		/** max command args */
-#define CONFIG_NO_RELOCATION		1
 
 /** System PLL reference clock */
 #define PLL_REF_CLK			50000000	/* 50 MHz */
@@ -435,7 +330,6 @@
 #define NS_PER_REF_CLK_TICK		(1000000000/PLL_REF_CLK)
 
 #define CONFIG_OCTEON_MMC
-#define CONFIG_CMD_MMC
 #define CONFIG_SYS_MMC_MAX_BLK_COUNT	8191
 #define CONFIG_SYS_MMC_SET_DEV
 
diff --git a/include/configs/thunderx_93xx.h b/include/configs/thunderx_93xx.h
index af5e1c4251..1af83e57da 100644
--- a/include/configs/thunderx_93xx.h
+++ b/include/configs/thunderx_93xx.h
@@ -7,9 +7,6 @@
 #ifndef __THUNDERX_93XX_H__
 #define __THUNDERX_93XX_H__
 
-#define CONFIG_SYS_VENDOR	"cavium"
-
-#define CONFIG_SPECIAL_SYNC_HANDLER
 
 /** Thunder 93xx does not support NOR flash */
 #define CONFIG_SYS_NO_FLASH
@@ -18,10 +15,6 @@
 
 #define CONFIG_SYS_BOOTM_LEN (16 << 20)
 
-/** Enable vsnprintf support */
-#define CONFIG_SYS_VSNPRINTF
-
-#define CONFIG_BOOTP_VCI_STRING		"Diagnostics"
 /* starting uboot and linux kernel from offset 40MB
  * below 40MB will be used for mailboxes of pko/pki
  * */
@@ -30,58 +23,9 @@
 
 #define CONFIG_COREID_MASK             0xffffff
 
-/** Enable full virtual address support */
-#define CONFIG_SYS_FULL_VA
-
 /** Low memory base address */
 #define CONFIG_SYS_LOWMEM_BASE		MEM_BASE
 
-/**
- * Memory map for virtual memory.  See arch/arm/include/asm/armv8/mmu.h
- * base:64, size:64, attributes:64
- */
-#define CONFIG_SYS_MEM_MAP		{{0x000000000000UL, 0x40000000000UL, \
-					  PTL2_MEMTYPE(MT_NORMAL) |	     \
-					  PTL2_BLOCK_NON_SHARE},	     \
-					 {0x800000000000UL, 0x40000000000UL, \
-					  PTL2_MEMTYPE(MT_DEVICE_NGNRNE) |   \
-					  PTL2_BLOCK_NON_SHARE},	     \
-					 {0x840000000000UL, 0x40000000000UL, \
-					  PTL2_MEMTYPE(MT_DEVICE_NGNRNE) |   \
-					  PTL2_BLOCK_NON_SHARE},	     \
-					 {0x880000000000UL, 0x40000000000UL, \
-					  PTL2_MEMTYPE(MT_DEVICE_NGNRNE) |   \
-					  PTL2_BLOCK_NON_SHARE},	     \
-					 {0xffffffffffffUL, 0xfffffffffffUL, \
-					  PTL2_MEMTYPE(MT_DEVICE_NGNRNE) |   \
-					  PTL2_BLOCK_NON_SHARE},	     \
-					}
-
-/** Number of entries in CONFIG_SYS_MEM_MAP */
-#define CONFIG_SYS_MEM_MAP_SIZE		4
-
-#define CONFIG_COREID_MASK		0xffffff
-
-/** Number of virtual address bits */
-#define CONFIG_SYS_VA_BITS		48
-
-/** Number of bits addressed by second level */
-#define CONFIG_SYS_PTL2_BITS		42
-
-/** Number of bits addressed by a single block entry from L2 page tables */
-#define CONFIG_SYS_BLOCK_SHIFT		29
-
-/** Number of 1st level page table entries */
-#define CONFIG_SYS_PTL1_ENTRIES		64
-
-/** Number of 2nd level page table entries */
-#define CONFIG_SYS_PTL2_ENTRIES		8192
-
-/** Total size of page table */
-#define CONFIG_SYS_PGTABLE_SIZE		\
-	((CONFIG_SYS_PTL1_ENTRIES + \
-	  CONFIG_SYS_MEM_MAP_SIZE * CONFIG_SYS_PTL2_ENTRIES) * 8)
-
 /** IPS field of TCR_EL1 */
 #define CONFIG_SYS_TCR_EL1_IPS_BITS	(5UL << 32)
 
@@ -138,9 +82,6 @@
  */
 #define CONFIG_SYS_MEM_TOP_HIDE		0x1000000
 
-/** Console serial port number */
-#define CONFIG_CONS_INDEX		1
-
 /* Generic Interrupt Controller Definitions */
 
 /** Base address of Global Interrupt Controller GICD registers */
@@ -179,12 +120,6 @@
 /** Maximum total number of BGX interfaces across all nodes */
 #define CONFIG_MAX_BGX			4
 
-/** Enable 10G PHY support */
-#define CONFIG_PHYLIB_10G
-#define CONFIG_PHY_VITESSE
-#define CONFIG_PHY_MICREL
-#define CONFIG_PHY_MARVELL
-
 /** Command line configuration */
 #define CONFIG_MENU
 
@@ -197,31 +132,12 @@
 /** Enable env command */
 #define CONFIG_CMD_ENV
 #undef  CONFIG_CMD_IMLS
-#define CONFIG_CMD_BOOTI
 
 #define CONFIG_CMD_MII
 
 /** Enable tftp command */
 #define CONFIG_CMD_TFTP
 
-/** Enable fatload, fatls, etc. */
-#define CONFIG_CMD_FAT
-
-/** Enable write support to FAT partitions */
-#define CONFIG_FAT_WRITE
-
-/** Enable ext2ls, ext2load, etc. commands */
-#define CONFIG_CMD_EXT2
-
-/** Enable ext4ls, ext4load, etc. commands */
-#define CONFIG_CMD_EXT4
-
-/** Enable part command */
-#define CONFIG_CMD_PART
-
-/** Enable sata command */
-#define CONFIG_CMD_SATA
-
 /** Enable pci command */
 #define CONFIG_CMD_PCI
 
@@ -234,18 +150,12 @@
 /** Enable grepenv command */
 #define CONFIG_CMD_GREPENV
 
-/** Enable bootimgup command */
-#define CONFIG_CMD_BOOTIMGUP
-
 /**
  * Enable callback support when environment variables are set, or
  * changed
  */
 #define CONFIG_CMD_ENV_CALLBACK
 
-/** Enable date command */
-#define CONFIG_CMD_DATE
-
 #define CONFIG_ENV_CALLBACK_LIST_STATIC "txsmi\\d?mode:smimode"
 
 /* AHCI support Definitions */
@@ -262,17 +172,6 @@
 # define CONFIG_SYS_64BIT_LBA
 #endif
 
-/* Partition systems */
-/** Enable old DOS partitioning for drives under 3TB */
-#define CONFIG_DOS_PARTITION
-/** Enable EFI partition support */
-#define CONFIG_EFI_PARTITION
-/** Enable UUIDs for drives >= 3TB */
-#define CONFIG_PARTITION_UUIDS
-
-/** Enable EXT4 filesystem support */
-#define CONFIG_FS_EXT4
-
 /* PCIe */
 /** Show devices found on PCI bus */
 #define CONFIG_PCI_SCAN_SHOW
@@ -300,18 +199,11 @@
 #define CONFIG_SYS_SPD_I2C_BUS 1
 
 /***** SPI Defines *********/
-#define CONFIG_DM_SPI
 #define CONFIG_DM_SPI_FLASH
-#define CONFIG_SPI_FLASH
-#define CONFIG_SPI_FLASH_STMICRO
-#define CONFIG_CAVIUM_SPI
 #define CONFIG_SF_DEFAULT_SPEED 12500000
 #define CONFIG_SF_DEFAULT_MODE	0
 #define CONFIG_SF_DEFAULT_BUS	0
 #define CONFIG_SF_DEFAULT_CS	0
-/** Enable sf command */
-#define CONFIG_CMD_SF
-
 /**************************/
 #define CONFIG_CMD_SAVES
 
@@ -321,7 +213,6 @@
 #define CONFIG_BOOTP_GATEWAY
 #define CONFIG_BOOTP_HOSTNAME
 #define CONFIG_BOOTP_PXE
-#define CONFIG_BOOTP_PXE_CLIENTARCH	0x100
 
 /* Miscellaneous configurable options */
 #define CONFIG_SYS_LOAD_ADDR		(MEM_BASE)
@@ -373,13 +264,10 @@
 /** Store U-Boot version in "ver" environment variable */
 #define CONFIG_VERSION_VARIABLE
 
-/** How long to delay before booting */
-#define CONFIG_BOOTDELAY		5
-
 /** Where the environment is located */
 
 /* Use ATF based calls for env variables */
-//#define CONFIG_ATF
+/*#define CONFIG_ATF*/
 
 /* Use uboot SPI APIs for env variables */
 #define CONFIG_SPI_ENV
@@ -436,7 +324,6 @@
 #define NS_PER_REF_CLK_TICK		(1000000000/PLL_REF_CLK)
 
 #define CONFIG_OCTEON_MMC
-#define CONFIG_CMD_MMC
 #define CONFIG_SYS_MMC_MAX_BLK_COUNT	8191
 #define CONFIG_SYS_MMC_SET_DEV
 
-- 
2.29.0

